### PR TITLE
Broken Fix for SponsorshipDiffTest cannot cancel sponsorship

### PR DESCRIPTION
**Pull request by xucito**
This fix only fixes the "senderNotIssuer" test however "insufficientReducedFee" is not working, the balance does not appear to be checked.

The fix itself doesn't make much sense as this is passing in waves.

### CODE REVIEW

**Comment by xucito** [comment]
Issue is now completely fixed. Underlying issue with double to int conversions caused with blockv5 making fees .5 of the base.

File: node/src/test/scala/com/wavesplatform/state/diffs/SponsorshipDiffTest.scala [hunk 1]
```diff
@@ -208,29 +208,29 @@ class SponsorshipDiffTest extends PropSpec with PropertyChecks with WithState wi
...
       ts         <- timestampGen
-      genesis1: GenesisTransaction = GenesisTransaction.create(master.toAddress, 1000000000000L, ts).explicitGet()
-      genesis2: GenesisTransaction = GenesisTransaction.create(notSponsor.toAddress, 1000000000000L, ts).explicitGet()
+      genesis1: GenesisTransaction = GenesisTransaction.create(master.toAddress, 2000000000000L, ts).explicitGet()
+      genesis2: GenesisTransaction = GenesisTransaction.create(notSponsor.toAddress, 2000000000000L, ts).explicitGet()
       (issueTx, sponsorTx, _, _) <- sponsorFeeCancelSponsorFeeGen(master)
       assetId = IssuedAsset(issueTx.id())
       senderNotIssuer = SponsorFeeTransaction
-        .selfSigned(1.toByte, notSponsor, assetId, None, 10 * Constants.UnitsInWave, ts + 1)
+        .selfSigned(1.toByte, notSponsor, assetId, None, 1 * Constants.UnitsInWave, ts + 1)
```
Comment (BlackTurtle123) [inline node/src/test/scala/com/wavesplatform/state/diffs/SponsorshipDiffTest.scala]: 500 * 2000000 = 10 00000000 = old fee

File: node/src/test/scala/com/wavesplatform/state/diffs/SponsorshipDiffTest.scala [hunk 2]
```diff
@@ -208,29 +208,29 @@ class SponsorshipDiffTest extends PropSpec with PropertyChecks with WithState wi
...
         .explicitGet()
       insufficientFee = SponsorFeeTransaction
-        .selfSigned(1.toByte, notSponsor, assetId, None, 10 * Constants.UnitsInWave - 1, ts + 1)
+        .selfSigned(1.toByte, notSponsor, assetId, None, 1 * Constants.UnitsInWave - 1, ts + 1)
```
Comment (BlackTurtle123) [inline node/src/test/scala/com/wavesplatform/state/diffs/SponsorshipDiffTest.scala]: Same here

File: node/src/test/scala/com/wavesplatform/state/diffs/SponsorshipDiffTest.scala [hunk 3]
```diff
@@ -241,27 +241,27 @@ class SponsorshipDiffTest extends PropSpec with PropertyChecks with WithState wi
...
-      case (genesis1, genesis2, issueTx, sponsorTx, senderNotIssuer, insufficientFee) =>
-        val setupBlocks = Seq(block(Seq(genesis1,genesis2,issueTx,sponsorTx)))
+      case (preconditions, senderNotIssuer, insufficientFee) =>
+        val setupBlocks = Seq(block(preconditions))
         assertDiffEi(setupBlocks, block(Seq(senderNotIssuer)), s) { blockDiffEi =>
           blockDiffEi should produce("Asset was issued by other address")
         }
         assertDiffEi(setupBlocks, block(Seq(insufficientFee)), s) { blockDiffEi =>
-          blockDiffEi should produce("(999999 in TN) does not exceed minimal value of 1000000 TN")
+          blockDiffEi should produce("(999999999 in TN) does not exceed minimal value of 1000000000 TN")
```
Comment (BlackTurtle123) [inline node/src/test/scala/com/wavesplatform/state/diffs/SponsorshipDiffTest.scala]: isn't this after blovkv5 activatie? so it should be lower?

**Pull request merged**